src/deadline/client/ui/dialogs/submit_job_progress_dialog.py

```diff
@@ -544,42 +591,61 @@ def handle_thread_exception(self, e: BaseException) -> None:
         logger.error(str(e))
 
     def _confirm_asset_references_outside_storage_profile(
-        self,
-        deviated_file_count_by_root: dict[str, int],
-        num_files: int,
-        upload_size: int,
+        self, upload_group: AssetUploadGroup
     ) -> bool:
         """
         Creates a dialog to prompt the user to confirm that they want to proceed
         with uploading when files were found outside of the configured storage profile locations.
         """
-        message_box = QMessageBox(self)
         message_text = (
-            f"Job submission contains {num_files} files totaling {_human_readable_file_size(upload_size)}. "
-            " All files will be uploaded to S3 if they are not already present in the job attachments bucket."
+            f"Job submission contains {upload_group.total_input_files} input files totaling {_human_readable_file_size(upload_group.total_input_bytes)}. "
+            " All input files will be uploaded to S3 if they are not already present in the job attachments bucket."
         )
-        if deviated_file_count_by_root:
-            root_by_count_message = "\n\n".join(
-                [
-                    f"{file_count} files from : '{directory}'"
-                    for directory, file_count in deviated_file_count_by_root.items()
-                ]
-            )
+        warning_message = ""
+        for group in upload_group.asset_groups:
+            if not group.file_system_location_name:
+                warning_message += f"\n\nUnder the directory '{group.root_path}':"
+                warning_message += (
+                    f"\n\t{len(group.inputs)} input file{'' if len(group.inputs) == 1 else 's'}"
+                    if len(group.inputs) > 0
+                    else ""
+                )
+                warning_message += (
+                    f"\n\t{len(group.outputs)} output director{'y' if len(group.outputs) == 1 else 'ies'}"
+                    if len(group.outputs) > 0
+                    else ""
+                )
+                warning_message += (
+                    f"\n\t{len(group.references)} referenced file{'' if len(group.references) == 1 else 's'} and/or director{'y' if len(group.outputs) == 1 else 'ies'}"
+                    if len(group.references) > 0
+                    else ""
+                )
+
+        # Exit early if we've set auto accept and there are no warnings
+        if not warning_message and self._auto_accept:
+            return True
+
+        # Build the UI
+        message_box = QMessageBox(self)
+        if warning_message:
             message_text += (
-                f"\n\nFiles were found outside of the configured storage profile location(s). "
-                " Please confirm that you intend to upload files from the following directories:\n\n"
-                f"{root_by_count_message}"
+                f"\n\nFiles were specified outside of the configured storage profile location(s). "
```

Comment:
Can we print out storage profile locations? It might be helpful to compare & notice problems when customers see this message

Reply:
Good callout! I wasn't able to update the CLI flow since the callback doesn't take the Storage Profile and I don't want to blow up the scope of this PR any more than necessary, but the GUI flow already had access to the Storage Profile so I've made your suggested change. Here's some screenshots (1) with SPs configured, (2) if there aren't any SPs:

<img width="521" alt="Screenshot 2024-04-25 at 12 59 48 PM" src="https://github.com/aws-deadline/deadline-cloud/assets/132690522/bccc8477-60e3-4f3a-93f7-05e4083b52bd">
<img width="516" alt="Screenshot 2024-04-25 at 12 59 23 PM" src="https://github.com/aws-deadline/deadline-cloud/assets/132690522/35ae41c4-f92b-42e2-9774-cc4d8483c563">